test/step-function.localstack.test.ts

```diff
@@ -370,6 +370,40 @@ localstackTestSuite("sfnStack", (testResource, _stack, _app) => {
     [1, 2]
   );
 
+  test(
+    "$AWS.SDK.CloudWatch",
+    (parent) => {
+      const alarm = new aws_cloudwatch.Alarm(parent, "Alarm", {
+        evaluationPeriods: 1,
+        threshold: 1,
+        metric: new aws_cloudwatch.Metric({
+          namespace: "AWS/Lambda",
+          metricName: "Errors",
+        }),
+      });
+
+      return new StepFunction<{}, string | undefined>(
+        parent,
+        "fn",
+        async () => {
+          const { MetricAlarms } = await $AWS.SDK.CloudWatch.describeAlarms(
+            {
+              AlarmNames: [alarm.alarmName],
+            },
+            { iamResources: [alarm.alarmArn] }
+          );
```

Comment:
Maybe this is isn't important, but I find this to not be aesthetically pleasing - 2 object literals. Would it be too risky to move these parameters into the payload itself?

Reply:
Or maybe:
```ts
$AWS.SDK.CloudWatch.withIAM({
  resources: [alarm.alarmArn]
}).describeAlarms({
  AlarmNames: [alarm.alarmName]
});
```

Reply:
I am definitely with you on this one. It just doesn't look good.

Messing with the sdk call input seems a bit risky as I wouldn't want to accidentally overwrite something in the input.